packages/expo-updates/android/src/main/java/expo/modules/updates/manifest/LegacyManifest.java

```diff
@@ -158,39 +158,43 @@ public UpdateEntity getUpdateEntity() {
 
   private Uri getAssetsUrlBase() {
     if (mAssetsUrlBase == null) {
-      String hostname = mManifestUrl.getHost();
-      if (hostname == null) {
-        mAssetsUrlBase = Uri.parse(EXPO_ASSETS_URL_BASE);
-      } else {
-        for (String expoDomain : EXPO_DOMAINS) {
-          if (hostname.contains(expoDomain)) {
-            mAssetsUrlBase = Uri.parse(EXPO_ASSETS_URL_BASE);
-            break;
-          }
+      mAssetsUrlBase = getAssetsUrlBase(mManifestUrl, getRawManifestJson());
+    }
+    return mAssetsUrlBase;
+  }
+
+  /* package */ static Uri getAssetsUrlBase(Uri manifestUrl, JSONObject manifestJson) {
+    String hostname = manifestUrl.getHost();
+    if (hostname == null) {
+      return Uri.parse(EXPO_ASSETS_URL_BASE);
+    } else {
+      for (String expoDomain : EXPO_DOMAINS) {
+        if (hostname.contains(expoDomain)) {
```

Comment:
This should be `equals expoDomain or ends with "." + expoDomain` (expo.io.uhoh.com)